aws/table_aws_spot_price_history.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+	"time"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsSpotPriceHistory(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_spot_price_history",
+		Description: "AWS Spot Price History",
+		List: &plugin.ListConfig{
+			Hydrate: listSpotPriceHistory,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "availability_zone", Require: plugin.Optional},
+				{Name: "instance_type", Require: plugin.Optional},
+				{Name: "product_description", Require: plugin.Optional},
+				{Name: "start_time", Require: plugin.Optional},
+				{Name: "end_time", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{Name: "availability_zone", Description: "The Availability Zone.", Type: proto.ColumnType_STRING},
+			{Name: "instance_type", Description: "The instance type.", Type: proto.ColumnType_STRING},
+			{Name: "product_description", Description: "A general description of the AMI.", Type: proto.ColumnType_STRING},
+			{Name: "spot_price", Description: "The maximum price per unit hour that you are willing to pay for a Spot Instance.", Type: proto.ColumnType_STRING},
+			{Name: "timestamp", Description: "The date and time the request was created", Type: proto.ColumnType_TIMESTAMP},
```

Comment:
This is a [non-reserved keyword in PostgreSQL](https://www.postgresql.org/docs/current/sql-keywords-appendix.html), so you can't create functions or types with its name, but it feels a bit risky to use this as a column name.

Would something like `create_timestamp` or `request_timestamp` make sense?